src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java

```diff
@@ -0,0 +1,407 @@
+                getBlobName(2, 0, compression),
+                getBlobName(3, 0, compression));
+        assertIterableEquals(expectedBlobs, testBucketAccessor.getBlobNames(gcsPrefix));
+
+        final Map<String, List<GenericRecord>> blobContents = new HashMap<>();
+        for (final String blobName : expectedBlobs) {
+            final var records =
+                    ParquetUtils.readRecords(
+                            tmpDir.resolve(Paths.get(blobName)),
+                            testBucketAccessor.readBytes(blobName)
+                    );
+            blobContents.put(blobName, records);
+        }
+        cnt = 0;
+        for (int i = 0; i < 10; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final var name = "user-" + cnt;
+                final String blobName = getBlobName(partition, 0, compression);
+                final var record = blobContents.get(blobName).get(i);
+                final String expectedLine = "{\"value\": {\"name\": \"" + name + "\"}}";
+                assertEquals(expectedLine, record.toString());
+                cnt += 1;
+            }
+        }
+    }
+
+    @Test
+    final void schemaChanged(@TempDir final Path tmpDir)
+            throws ExecutionException, InterruptedException, IOException {
+        final Map<String, String> connectorConfig = basicConnectorConfig();
+        final var compression = "none";
+        connectorConfig.put("format.output.fields", "value");
+        connectorConfig.put("format.output.fields.value.encoding", "none");
+        connectorConfig.put("key.converter", "org.apache.kafka.connect.storage.StringConverter");
+        connectorConfig.put("value.converter", "org.apache.kafka.connect.json.JsonConverter");
+        connectorConfig.put("file.compression.type", compression);
+        connectorConfig.put("format.output.type", "parquet");
+        connectRunner.createConnector(connectorConfig);
+
+        final var jsonMessageSchema = "{\"type\":\"struct\",\"fields\":[{\"type\":\"string\",\"field\":\"name\"}]}";
+        final var jsonMessageNewSchema = "{\"type\":\"struct\",\"fields\":"
+                + "[{\"type\":\"string\",\"field\":\"name\"}, "
+                + "{\"type\":\"string\",\"field\":\"value\", \"default\": \"foo\"}]}";
+        final var jsonMessagePattern = "{\"schema\": %s, \"payload\": %s}";
+
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        final var expectedRecords = new ArrayList<String>();
+        int cnt = 0;
+        for (int i = 0; i < 10; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final var key = "key-" + cnt;
+                final var payload = i < 5
+                        ? "{" + "\"name\": \"user-" + cnt + "\"}"
+                        : "{" + "\"name\": \"user-" + cnt + "\", \"value\": \"value-" + cnt + "\"}";
+                final String value = i < 5
+                        ? String.format(
```

Comment:
We could use ternary operator for jsonMessageSchema only. It looks like everything else stays the same

Reply:
better `if/else` as result it is readble